fhir-search/src/main/java/com/ibm/fhir/search/util/ReferenceValue.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.search.util;
+
+
+/**
+ * A reference extracted from a FHIR resource, with processing applied to
+ * identify the reference type and target resource type
+ */
+public class ReferenceValue {
+
+    public static enum ReferenceType {
+        LITERAL_RELATIVE, // Patient/123abc
+        LITERAL_ABSOLUTE, // http(s)://an.other.server/Patient/234def
+        LOGICAL,          // e.g. SSN
+        DISPLAY_ONLY,     // A Reference with only a display field
+        INVALID           // Not a valid reference
+    }
```

Comment:
I like having the single reference I can go back to and see - this was a good addition and comments.